### DISCUSSION TITLE
What is the SQS (HTTP) default timeout?

### DISCUSSION THREAD

**Discussion by snspinn** [awslabs/aws-sdk-rust#1077] [Q&A] [answered]
What is the HTTP timeout or the underlying HTTP library so that I can check?

The `aws_sdk_sqs::Client` docs don't really say (why does it talk about the Java library?).

> To avoid HTTP errors, ensure that the HTTP response timeout for ReceiveMessage requests is longer than the WaitTimeSeconds parameter. For example, with the Java SDK, you can set HTTP transport settings using the[ NettyNioAsyncHttpClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.html) for asynchronous clients, or the[ ApacheHttpClient](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.html) for synchronous clients.

**Comment by snspinn** [comment]
Just trying some values and get this:

```
Receive error: ServiceError(ServiceError { source: Unhandled(Unhandled { source: ErrorMetadata { code: Some("InvalidParameterValue"), message: Some("Value 120 for parameter WaitTimeSeconds is invalid. Reason: Must be >= 0 and <= 20, if provided."), extras: Some({"type": "Sender", "aws_request_id": "ff7566ef-52ef-59c6-a0d6-38b76d99f5cf"}) }, meta: ErrorMetadata { code: Some("InvalidParameterValue"), message: Some("Value 120 for parameter WaitTimeSeconds is invalid. Reason: Must be >= 0 and <= 20, if provided."), extras: Some({"type": "Sender", "aws_request_id": "ff7566ef-52ef-59c6-a0d6-38b76d99f5cf"}) } }), raw: Response { status: StatusCode(400), headers: Headers { headers: {"x-amzn-requestid": HeaderValue { _private: H0("ff7566ef-52ef-59c6-a0d6-38b76d99f5cf") }, "x-amzn-query-error": HeaderValue { _private: H0("InvalidParameterValue;Sender") }, "date": HeaderValue { _private: H0("Tue, 27 Feb 2024 19:51:18 GMT") }, "content-type": HeaderValue { _private: H0("application/x-amz-json-1.0") }, "content-length": HeaderValue { _private: H0("177") }, "connection": HeaderValue { _private: H0("keep-alive") }} }, body: SdkBody { inner: Once(Some(b"{\"__type\":\"com.amazon.coral.service#InvalidParameterValueException\",\"message\":\"Value 120 for parameter WaitTimeSeconds is invalid. Reason: Must be >= 0 and <= 20, if provided.\"}")), retryable: true }, extensions: Extensions { extensions_02x: Extensions, extensions_1x: Extensions } } })
```

Guys, I just realised that the quoted section in my previous comment is just a copy and paste from the main AWS docs (which are infamous in some circles). Why clutter the crate docs with this useless boilerplate??

Surely these input parameters could be compile timed checked? Please don't drag this Rust library down to the level of a JS one :pray:

**Reply by ysaito1001** [reply, marked as answer]
Thank you for your query. These docs are auto-generated from a service model ([example](https://github.com/awslabs/aws-sdk-rust/blob/72eb93fe4be5951d6351b48df850d2be1f29edf6/aws-models/sqs.json)). You are right in that it is weird the rust docs mention the Java SDK; this directly comes from the documentation the SQS team and appears in docs of other languages' SDK. 
 
Similarly, `aws_sdk_sqs` itself is pretty much code-generated by [smithy-rs](https://github.com/smithy-lang/smithy-rs) and input parameter validation is currently not present in the generated code.
